src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -1,184 +1,115 @@
 package sbtcrossproject
 
-import scala.language.implicitConversions
-import scala.language.experimental.macros
-
-import scala.reflect.macros.Context
-
-import sbt._
-import Keys._
-import Project.projectToRef
-
 import java.io.File
 
-final class CrossProject private (
-    crossType: CrossType,
-    val projects: Map[CrossPlatform, Project]
-) {
-  import CrossProject._
-
-  // Transformers for inner projects
-
-  def configure(platform: CrossPlatform, transform: Project => Project) =
-    copy(projects = projects + (platform -> transform(projects(platform))))
-
-  def settings(platform: CrossPlatform, ss: Def.Setting[_]*): CrossProject =
-    configure(platform, _.settings(ss: _*))
-
-  // Scala.js-style cross project aliases
-
-  /** Project for JVM platform. */
-  def jvm: Project = projects(JVM)
-
-  /** Transform the underlying JVM project */
-  def jvmConfigure(transform: Project => Project): CrossProject =
-    configure(JVM, transform)
+import sbt.{AutoPlugin, Configuration, Plugins, Project, ProjectReference}
 
-  /** Add settings specific to the underlying JVM project */
-  def jvmSettings(ss: Def.Setting[_]*): CrossProject =
-    settings(JVM, ss: _*)
-
-  // Concrete alteration members
-
-  def aggregate(refs: CrossProject*): CrossProject =
-    mapCrossPlatforms { p =>
-      projects(p).aggregate(refs.map(_.projects(p): ProjectReference): _*)
-    }
-
-  def configs(cs: Configuration*): CrossProject =
-    mapProjects(_.configs(cs: _*))
+import scala.language.experimental.macros
+import scala.language.implicitConversions
+import scala.reflect.macros.Context
 
-  def configureCross(transforms: (CrossProject => CrossProject)*): CrossProject =
-    transforms.foldLeft(this)((p, t) => t(p))
+/**
+ * Created by grinder on 17.07.16.
+ */
+class CrossProject[A <: Platform] private (
+    id: String,
+    root: File,
+    val projects: Map[Platform, Project]) {
+  def settings(t: Platform,
+               settings: Seq[_root_.sbt.Def.Setting[_]]): CrossProject[A] =
+    copy(projects + (t -> projects(t).settings(settings)))
```

Comment:
This method should not be used in most cased by end-users. It is necessary to build pimps like `.jvmSettings`, but in the end user build, there should be very few cases where calling this method is a good idea. Therefore, it should have a name more obscure than just `settings`. Maybe it could even be put in a separate object to hide it further.